src/ac_copilot_trainer/modules/realtime_coaching.lua

```
@@ -0,0 +1,302 @@
+-- O(1) spline lookup via quantized bucket index for Nordschleife (170+ segments).
+
+local coachingHints = require("coaching_hints")
+
+local M = {}
+
+-- ---------------------------------------------------------------------------
+-- Constants
+-- ---------------------------------------------------------------------------
+
+local NUM_BUCKETS = 1000          -- quantization resolution for O(1) lookup
+local EXIT_WINDOW_SPLINE = 0.008  -- spline distance past corner s1 for "exiting" phase
```

Luacheck (1.2.0)

[warning] unused variable 'EXIT_WINDOW_SPLINE'

(W211)